packages/flutter/lib/src/widgets/text.dart

```diff
@@ -636,6 +640,16 @@ class Text extends StatelessWidget {
   /// (semi-transparent grey).
   final Color? selectionColor;
 
+  /// Controls how tall the selection highlight boxes are computed to be.
+  ///
+  /// See [ui.BoxHeightStyle] for details on available styles.
+  final ui.BoxHeightStyle selectionHeightStyle;
```

Comment:
I think we should probably add some snippet about how these values only work under a `SelectionArea`. It does seem a bit strange to me that `Text` will have some parameters that are only valid under a `SelectionArea`. Maybe we could extract these parameters into some selection configuration class?

Reply:
From text input triage: maybe this can be exposed through `SelectionArea/SelectableRegion` and exposed to `Text` through some inherited widget.